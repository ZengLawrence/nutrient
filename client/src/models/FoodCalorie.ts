export interface FoodCalorie {
    _id?: string, 
    food? : string,
    caloriesPer100g? : number
}