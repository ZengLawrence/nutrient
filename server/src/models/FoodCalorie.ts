import mongoose from 'mongoose';

export interface IFoodCalorie extends mongoose.Document { 
    food: string,
    foodGroup: string,
    caloriesPer100g: number
}

export const FoodCalorieSchema = new mongoose.Schema({
    food: { type: String, required: true},
    foodGroup: { type: String, required: true},
    caloriesPer100g: { type: Number, required: true}
});

const FoodCalorie = mongoose.model<IFoodCalorie>("FoodCalorie", FoodCalorieSchema);
export default FoodCalorie;