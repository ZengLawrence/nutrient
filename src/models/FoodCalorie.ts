import mongoose from 'mongoose';

export interface IFoodCalorie extends mongoose.Document { 
    food: string,
    foodGroup: string,
    caloriesPer100g: number
}

const foodCalorieSchema = new mongoose.Schema({
    food: { type: String, required: true},
    foodGroup: { type: String, required: true},
    caloriesPer100g: { type: Number, required: true}
});
foodCalorieSchema.set('toJSON', { virtuals: true });

const FoodCalorie = mongoose.model<IFoodCalorie>("FoodCalorie", foodCalorieSchema);
export default FoodCalorie;