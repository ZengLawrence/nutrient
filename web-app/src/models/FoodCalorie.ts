export interface FoodCalorie { 
    food? : string,
    foodGroup?: string,
    caloriesPer100g? : number
}