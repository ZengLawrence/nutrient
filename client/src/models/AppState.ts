import { FoodCalorie } from './FoodCalorie';

export interface AppState {
    foodCalories: FoodCalorie[],
}