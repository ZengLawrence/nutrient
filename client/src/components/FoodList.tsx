import * as React from 'react';
import { FoodCalorie } from '../models/FoodCalorie';
import { Table } from 'reactstrap';

export interface FoodListProps {
    foodCalories: FoodCalorie[]
}

export const FoodList = (props: FoodListProps) => {
    return (
        <Table hover>
            <thead>
                <tr className="text-center p-3 mb-2 bg-light">
                    <th className="col-sm-8">Food</th>
                    <th className="col-sm-4">Calories/100g</th>
                </tr>
            </thead>
            <tbody>
                {props.foodCalories.map(item => (
                    // Without the `key`, React will fire a key warning
                    <React.Fragment key={item._id}>
                        <tr>
                            <td>{item.food}</td>
                            <td className="text-right">{item.caloriesPer100g}</td>
                        </tr>
                    </React.Fragment>
                ))}
            </tbody>
        </Table>
    );
};

