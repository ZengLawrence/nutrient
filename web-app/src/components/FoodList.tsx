import * as React from 'react';
import { FoodCalorie } from '../models/FoodCalorie';

export interface FoodListProps {
    foodCalories: FoodCalorie[]
}

export const FoodList = (props: FoodListProps) => {
    return (
        <table>
            <thead>
                <tr>
                    <th>Food</th>
                    <th>Food Group</th>
                    <th>Calries per 100g</th>
                </tr>
            </thead>
            <tbody>
                {props.foodCalories.map(item => (
                    // Without the `key`, React will fire a key warning
                    <tr>
                        <React.Fragment key={item.food}>
                            <td>{item.food}</td>
                            <td>{item.foodGroup}</td>
                            <td>{item.caloriesPer100g}</td>
                        </React.Fragment>
                    </tr>
                ))}
            </tbody>
        </table>
    );
}

