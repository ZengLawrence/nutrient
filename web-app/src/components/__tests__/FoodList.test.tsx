import React from 'react';
import { render } from '@testing-library/react';
import { FoodList } from '../FoodList';
import { FoodCalorie } from '../../models/FoodCalorie';

it('renders empty food calorie list', () => {
  const { asFragment } = render(<FoodList foodCalories={[]} />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders food calorie list with 3 food items', () => {
    const foodCalories: FoodCalorie[] = [
        {food: 'Orange', foodGroup: 'Fruit', caloriesPer100g: 47},
        {food: 'Sliced Bread', foodGroup: 'Carbohydrate', caloriesPer100g: 90},
        {food: 'Chicken Breast', foodGroup: 'Protein', caloriesPer100g: 110},
    ];
    const { asFragment } = render(<FoodList foodCalories={foodCalories} />);
    expect(asFragment()).toMatchSnapshot();
  });