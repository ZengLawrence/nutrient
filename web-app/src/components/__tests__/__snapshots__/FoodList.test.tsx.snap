// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty food calorie list 1`] = `
<DocumentFragment>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Food Group
        </th>
        <th>
          Food
        </th>
        <th>
          Calries per 100g
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</DocumentFragment>
`;

exports[`renders food calorie list with 3 food items 1`] = `
<DocumentFragment>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Food Group
        </th>
        <th>
          Food
        </th>
        <th>
          Calries per 100g
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Fruit
        </td>
        <td>
          Orange
        </td>
        <td>
          47
        </td>
      </tr>
      <tr>
        <td>
          Carbohydrate
        </td>
        <td>
          Sliced Bread
        </td>
        <td>
          90
        </td>
      </tr>
      <tr>
        <td>
          Protein
        </td>
        <td>
          Chicken Breast
        </td>
        <td>
          110
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
